Extension/src/Debugger/configurationProvider.ts

```diff
@@ -591,40 +577,115 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
+                case TaskConfigStatus.configured : {
+                    item.detail = localize("configured.task", "Configured Task");
+                    break;
+                }
+                case TaskConfigStatus.detected : {
+                    item.detail = localize("detected.task", "Detected Task");
+                    break;
+                }
+                default : {
+                    break;
+                }
+            }
+            if (item.configuration.taskDetail) {
+                // Add the compiler path of the preLaunchTask to the description of the debug configuration.
+                item.detail = (item.detail ?? "") + " (" + item.configuration.taskDetail + ")";
+            }
+        });
+        return items;
+    }
+
+    private findDefaultConfig(configs: CppDebugConfiguration[]): CppDebugConfiguration[] {
+        return configs.filter((config: CppDebugConfiguration) => (config.hasOwnProperty("isDefault") && config.isDefault));
+    }
+
+    private isDebugPanelConfig(config: CppDebugConfiguration): boolean {
+        return (config.debuggerEvent && config.debuggerEvent === DebuggerEvent.debugPanel) ||
+            (config.detail === undefined && config.source === undefined && config.taskStatus === undefined);
     }
 
-    public checkDebugConfigExists(configName: string, folder?: vscode.WorkspaceFolder, type?: DebuggerType): boolean {
-        const configs: vscode.DebugConfiguration[] | undefined = this.getLaunchConfigs(folder, type);
+    private isExistingConfig(config: CppDebugConfiguration, folder?: vscode.WorkspaceFolder, searchLaunchJsonOnly: boolean = false): boolean {
+        if (this.isDebugPanelConfig(config) || (!searchLaunchJsonOnly && config.taskStatus && (config.taskStatus !== TaskConfigStatus.detected))) {
+            return true;
+        } else if (config.taskStatus && (config.taskStatus === TaskConfigStatus.detected)) {
+            return false;
+        }
+
+        const configs: CppDebugConfiguration[] | undefined = this.getLaunchConfigs(folder, config.type);
         if (configs && configs.length > 0) {
-            const selectedConfig: any | undefined = configs.find((config: any) => config.name && config.name === configName);
+            const selectedConfig: any | undefined = configs.find((item: any) => item.name && item.name === config.name);
             if (selectedConfig) {
                 return true;
             }
         }
         return false;
     }
 
+    private getDebugConfigSource(config: CppDebugConfiguration, folder?: vscode.WorkspaceFolder): DebugConfigSource | undefined {
+        if (config.source) {
+            return config.source;
+        }
+        const configs: CppDebugConfiguration[] | undefined = this.getLaunchConfigs(folder, config.type);
+        const matchingConfig: CppDebugConfiguration | undefined = configs?.find((item: any) => item.name && item.name === config.name);
+        if (matchingConfig && matchingConfig.source) {
```

Comment:
Simplify `if (matchingConfig && matchingConfig.source)` to `if (matchingConfig?.source)` ?